Automattic/woocommerce-payments
# BNPL - Fix sending of shipping data for Afterpay in classic checkout

Fixes part of #6792 

#### Changes proposed in this Pull Request

Make sure for Afterpay payments, the shipping address is sent when both billing and shipping address are separately mentioned at checkout. So far, only the billing address was sent, even if there was a shipping address separately specified. 

The PR also does the following refactor: 

* Remove redundant PHP code that attempted to send shipping address when payment method was Afterpay
* Move`getShippingDetails` and `getBillingDetails` methods to Utils file so that it can be used in both Legacy and Split UPE versions without repeating the same code. 

There will be a separate PR for blocks checkout. 

#### Testing instructions

1.  Make sure your local setup is connected ot an account with Afterpay enabled
2. Within WooPayments Dev tools make sure to select `Legacy UPE` 
3. Open the site in an incognito window and try to do a checkout in classic checkout with Afterpay
4. Add only the Billing address without opting to add a separate shipping address
5. Open the browser console to monitor any errors ( console and network tab ) 
6. Complete the checkout and wait for the test redirect screen of Stripe to appear that shows the data submitted in the transaction
7. Within the Payment Intent details on that page, please check the Shipping address, it should be the billing address
8. Repeat the test transaction steps - this time - add a separate Shipping address.
9. The submitted Payment intent information on the test Afterpay authorization page should show the specified Billing address , and not the shipping address.

-------------------


**Post merge**
